bin/node/src/chain_spec/builder.rs

```diff
@@ -0,0 +1,164 @@
+use std::string::ToString;
+
+use aleph_runtime::{Feature, Perbill, WASM_BINARY};
+use pallet_staking::{Forcing, StakerStatus};
+use primitives::{
+    staking::{MIN_NOMINATOR_BOND, MIN_VALIDATOR_BOND},
+    AccountId, AlephNodeSessionKeys as SessionKeys, Version as FinalityVersion, ADDRESSES_ENCODING,
+    TOKEN_DECIMALS,
+};
+use serde_json::{Number, Value};
+
+use crate::{
+    chain_spec::{cli::ChainParams, AlephNodeChainSpec},
+    commands::AuthorityKeys,
+};
+
+fn to_account_ids(authorities: &[AuthorityKeys]) -> impl Iterator<Item = AccountId> + '_ {
+    authorities.iter().map(|auth| auth.account_id.clone())
+}
+
+fn system_properties(token_symbol: String) -> serde_json::map::Map<String, Value> {
+    [
+        ("tokenSymbol".to_string(), Value::String(token_symbol)),
+        (
+            "tokenDecimals".to_string(),
+            Value::Number(Number::from(TOKEN_DECIMALS)),
+        ),
+        (
+            "ss58Format".to_string(),
+            Value::Number(Number::from(ADDRESSES_ENCODING)),
+        ),
+    ]
+    .iter()
+    .cloned()
+    .collect()
+}
+
+/// Generate chain spec for new AlephNode chains
+pub fn build_chain_spec(
+    chain_params: ChainParams,
+    authorities: Vec<AuthorityKeys>,
+) -> Result<AlephNodeChainSpec, String> {
+    let token_symbol = String::from(chain_params.token_symbol());
+    let sudo_account = chain_params.sudo_account_id();
+    let rich_accounts = chain_params.rich_account_ids();
+    let faucet_account = chain_params.faucet_account_id();
+    let finality_version = chain_params.finality_version();
+
+    Ok(AlephNodeChainSpec::builder(
+        WASM_BINARY.ok_or("AlephNode development wasm not available")?,
+        Default::default(),
+    )
+    .with_name(chain_params.chain_name())
+    .with_id(chain_params.chain_id())
+    .with_chain_type(chain_params.chain_type())
+    .with_genesis_config_patch(generate_genesis_config(
+        authorities.clone(),    // Initial PoA authorities, will receive funds
+        sudo_account.clone(),   // Sudo account, will also be pre funded
+        rich_accounts.clone(),  // Pre-funded accounts
+        faucet_account.clone(), // Pre-funded faucet account
+        finality_version,
+    ))
+    .with_properties(system_properties(token_symbol))
+    .build())
+}
+
+/// Calculate initial endowments such that total issuance is kept approximately constant.
+fn calculate_initial_endowment(accounts: &[AccountId]) -> u128 {
+    let total_issuance = 300_000_000u128 * 10u128.pow(TOKEN_DECIMALS);
+    // due to known issue https://github.com/paritytech/polkadot-sdk/pull/2987/files,
+    // we need to make sure returned number is un u64 range, otherwise serde_json::json macro fails
+    // this is fixed in polkadot-sdk 1.6.0
+    total_issuance / (accounts.len() as u128) / 10
+}
+
+/// Configure initial storage state for FRAME modules.
+#[allow(clippy::too_many_arguments)]
```

Comment:
Maybe you could fix this since you are moving the code – or is there a reason why the arguments cannot be consolidated into a struct (or several)?

Also, are you sure this is even necessary? 5 arguments seems a bit low to trigger this check.

Reply:
Yes, I forgot to remove it, good point